src/Engine/Utils.java

```diff
@@ -681,16 +694,26 @@ public static boolean enqueueMoveEvent(MapMaster gameMap, Unit unit, GamePath mo
    * @param unit The unit who is to die.
    * @param eventQueue Will be given the new MoveEvent(s).
    */
-  public static void enqueueDeathEvent(Unit unit, GameEventQueue eventQueue)
+  public static void enqueueDeathEvent(Unit unit, XYCoord grave, boolean canLose, GameEventQueue eventQueue)
```

Comment:
`canLose` is never used in this function. Though if we were stomping our own unit, wouldn't the new unit then exist and prevent a loss without needing a flag to prevent it? I don't know that we have any relevant cases to test this.

Reply:
Goood point. Dunno how I missed that.
The new unit doesn't exist until the event finishes, while the `ArmyDefeatEvent` (once queued) doesn't care whether you've got units - you're just gonna lose.
I guess I probably should add a testcase or two.

Reply:
While we're talking deathy edge cases, consider:
Tank shoots my last unit
My last unit, on death, produces a new unit via event listener.
`ArmyDefeatEvent` still goes through and I lose.

This, I think, highlights the utility of pushing victory/defeat conditions out of event generation and into a bit of code that gets run when the event queue is empty.

Reply:
Yeah, I can see the logic for that. Just check victory/defeat conditions after each action or something. Probably less efficient, because we know what conditions to check (and for whom) right now, but simpler.

Reply:
Could even get fancy and add flags to know which conditions to check. No units died? Don't need to check if we are out of units. No buildings captured? No need to check if it was an HQ, or if there are none left, etc.

Some conditions would be harder to check ex post facto though; if an HQ gets captured, and I have another one, it's not obvious I should lose without knowing that history.

Reply:
OTOH, we're already tracking all HQ locations for each `Army`, so that's probably not a hard check to write... though that would result in the occasional false positive if there's no HQ/Lab (with the current codebase)

We will probably want flags in the `GameRules` that determine whether those things are loss conditions, though.